src/clims/services/workflow/camunda/client.py

```diff
@@ -0,0 +1,370 @@
+from __future__ import absolute_import
+
+import StringIO
+import logging
+import requests
+from lxml import etree
+from simplejson import JSONDecodeError
+from six.moves.urllib.parse import urljoin
+
+from .api import CamundaApi, UnexpectedHttpResponse
+from clims import utils
+from clims.models import Workflow
+
+logger = logging.getLogger(__name__)
+
+
+class CamundaClient(object):
+    """
+    A client for executing workflows that interfaces with the underlying workflow engine.
+    It's higher level than CamundaApi which directly maps to the CamundaApi.
+    """
```

Comment:
Right. The retry library was only brought in for tests. I think if we want to enable retries (which I think is a very good thing of course) for communication with external resources, we might perhaps want to do that with a message server?

I'll add a ticket for this, great point!

Reply:
But I'd also like to add that the Camunda server is going to be under little load in most setups (unless someone wants to use this to host a lims system for several organizations at once, which doesn't need to be in scope for now I guess). And the server is also running locally, so I'd argue that retries are not as important here as they would be if this was "truly" an external system. But I still think we should discuss it!

Reply:
No, I agree it might not be super relevant here. I just wanted to bring it since it is my experience that in any distributed system you will have these type of issues, even if it is all running on the same machine, etc. However, of course the larger the "distance" between systems, the higher the risk.

Reply:
Yes, I think it's a good point.